ophidian/parser/Lef.cpp

```diff
@@ -60,166 +56,174 @@ LefParser::~LefParser() {
 
 }
 
-std::shared_ptr<Lef> LefParser::readFile(const std::string &filename) {
-	return std::make_shared<Lef>(filename);
-}
+std::unique_ptr<Lef> LefParser::readFile(const std::string &filename) {
+	std::FILE* fp = std::fopen(filename.c_str(), "r");
+
+	if (!fp) {
+		// TODO: Change to proper exception when #41 is merged
+		throw "File could not be opened";
```

Comment:
Leak!!

Remember to `std::fclose(fp)`. I suggest you to use a `std::unique_ptr<std::FILE>` with a custom deleter.

read: http://stackoverflow.com/questions/26360916/using-custom-deleter-with-unique-ptr